libraries/AP_InertialSensor/AP_InertialSensor_L3G4200D.cpp

```diff
@@ -46,72 +48,115 @@ const extern AP_HAL::HAL &hal;
+#define L3G4200D_REG_WHO_AM_I                           0x0f
 #define L3G4200D_REG_WHO_AM_I_VALUE                     0xd3
 
-#define L3G4200D_REG_CTRL_REG1                     0x20
+#define L3G4200D_REG_CTRL_REG1                          0x20
 #define L3G4200D_REG_CTRL_REG1_DRBW_800_110             0xf0
 #define L3G4200D_REG_CTRL_REG1_PD                       0x08
 #define L3G4200D_REG_CTRL_REG1_XYZ_ENABLE               0x07
 
-#define L3G4200D_REG_CTRL_REG4                     0x23
+#define L3G4200D_REG_CTRL_REG4                          0x23
 #define L3G4200D_REG_CTRL_REG4_FS_2000                  0x30
 
-#define L3G4200D_REG_CTRL_REG5                     0x24
+#define L3G4200D_REG_CTRL_REG5                          0x24
 #define L3G4200D_REG_CTRL_REG5_FIFO_EN                  0x40
-
-#define L3G4200D_REG_FIFO_CTL                      0x2e
+#define L3G4200D_REG_FIFO_CTL                           0x2e
 #define L3G4200D_REG_FIFO_CTL_STREAM                    0x40
 
-#define L3G4200D_REG_FIFO_SRC                      0x2f
+#define L3G4200D_REG_FIFO_SRC                           0x2f
 #define L3G4200D_REG_FIFO_SRC_ENTRIES_MASK              0x1f
 #define L3G4200D_REG_FIFO_SRC_EMPTY                     0x20
 #define L3G4200D_REG_FIFO_SRC_OVERRUN                   0x40
 
-#define L3G4200D_REG_XL                            0x28
+#define L3G4200D_REG_XL                                 0x28
 
 // this bit is ORd into the register to enable auto-increment mode
-#define L3G4200D_REG_AUTO_INCREMENT		           0x80
+#define L3G4200D_REG_AUTO_INCREMENT		                0x80
 
 // L3G4200D Gyroscope scaling
 // running at 2000 DPS full range, 16 bit signed data, datasheet
 // specifies 70 mdps per bit
 #define L3G4200D_GYRO_SCALE_R_S (DEG_TO_RAD * 70.0f * 0.001f)
 
 // constructor
-AP_InertialSensor_L3G4200D::AP_InertialSensor_L3G4200D(AP_InertialSensor &imu,
-                                                       AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev)
+AP_InertialSensor_L3G4200D_2::AP_InertialSensor_L3G4200D_2(AP_InertialSensor &imu,
+                                                    AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev_gyro,
+                                                    AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev_accel)
     : AP_InertialSensor_Backend(imu)
-    , _dev(std::move(dev))
+    , _dev_gyro(std::move(dev_gyro))
+    , _dev_accel(std::move(dev_accel))
 {
 }
 
-AP_InertialSensor_L3G4200D::~AP_InertialSensor_L3G4200D()
+AP_InertialSensor_L3G4200D_2::AP_InertialSensor_L3G4200D_2(AP_InertialSensor &imu,
+                                                       AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev_gyro,
+                                                       AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev_accel,
+                                                       enum Rotation rotation)
+
+    : AP_InertialSensor_Backend(imu)
+    , _rotation(rotation)
+    , _dev_gyro(std::move(dev_gyro))
+    , _dev_accel(std::move(dev_accel))
+{
+}
```

Comment:
no need for the second constructor... this can only be used with both accel and gyro, so you should actually replace the first one.

Reply:
true ... I have moved it. I believe it should be added later and call 
   ```
// setup sensor rotations from probe()
    set_gyro_orientation(_gyro_instance, _rotation);
    set_accel_orientation(_accel_instance, _rotation);
```
as in [AP_InertialSensor_Invensensev2.cpp L229
](https://github.com/HefnySco/ardupilot/blob/pr_Updating_AP_InertialSensor_L3G4200D_part2/libraries/AP_InertialSensor/AP_InertialSensor_Invensensev2.cpp#L229)